Add changelog for 1.0.0-beta.1 (#2175)
jupyterhub/zero-to-jupyterhub-k8s: CHANGELOG.md

Here you can find upgrade changes in between releases and upgrade instructions.

## [12.0] - 2021-05-FIXME

### [12.0.0]

FIXME: summarize release

- note 12.0.0 bump from 0.11.1

#### Security announcement

FIXME: describe AKS cluster setup documentation issue.

#### Breaking changes

> - **Kubernetes 1.17+ and Helm 3.1+ is now required**

Reviewer: ```suggestion
- **Kubernetes 1.17+ and Helm 3.1+ are now required**
```

Here you can find upgrade changes in between releases and upgrade instructions.

## [1.0]

### [1.0.0] - 2021-05-FIXME

FIXME: Summary

- note 1.0.0 bump from 0.11.1
- arm64
- pullOnlyOnChanges
- extraFiles
- fullnameOverride
- seeding of secrets

  NOTE about the need to do one upgrade before removing secrets.

#### Security announcement

FIXME: describe AKS cluster setup documentation issue.

#### Breaking changes

> - **Kubernetes 1.17+ and Helm 3.1+ are now required**

Reviewer 2: ```suggestion
- **Kubernetes 1.17+ and Helm 3.5+ are now required**
```

  Helm 3 (3.5+) is now required. Helm 2 reached end of life last year and we

>   have started relying on Helm 3 specific features.

Reviewer 2: ```suggestion
  have started relying on Helm 3.5 specific features.
```

  have started relying on Helm 3.5 specific features.

  Kubernetes 1.17+ is now required. It helped us avoid maintaining two separate
  sets of implementations for the the user-scheduler.

- **Schema validation of chart config** ([#2033](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/2033), [#2200](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/2200))

  The Helm chart now bundles with a `values.schema.json` file that will validate
  all use of the Helm chart during template rendering. If the Helm chart's
  passed values doesn't comply with the schema, then `helm` will error before
  the k8s api-server has become involved and anything has changed in the k8s
  cluster.

  The most common validation errors are:

  - _Unrecognized config values_

    For example if you have misspelled something.

    Note that if you want to pass your custom values for inspection by custom
    logic in the hub pod, then you should pass these values via the `custom`
    config section where anything will be accepted.

  - _Recognized config values with the wrong type_

    For example if you have passed a numerical value to a configuration that
    expected a string.

> - default requests:
>   https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/2034
> 
>   prePuller.resources (0, 0)
>   prePuller.hook.resources (0, 0)
>   scheduling.userScheduler.resources (50m, 256Mi)
>   hub.resources (200m, 512Mi)
>   proxy.resources (200m, 512Mi)

Reviewer 2: ```suggestion
- **Default resource requests are no longer set** [#2034](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/2034)

  Z2JH now follows Helm chart best practice in not setting default resource limits.
  If you want to restore the old behaviour you can set:
  ```yaml
  hub:
    resources:
      requests:
        cpu: 200m
        memory: 512Mi

  proxy:
    chp:
      resources:
        requests:
          cpu: 200m
          memory: 512Mi

  scheduling:
    userScheduler:
      resources:
        requests:
          cpu: 50m
          memory: 256Mi

  prePuller:
    resources:
      requests:
        cpu: 0
        memory: 0
    hook:
      resources:
        requests:
          cpu: 0
          memory: 0
```
As mentioned on https://github.com/jupyterhub/zero-to-jupyterhub-k8s/issues/2085#issuecomment-846462083 I don't think we need to block this release on having full documentation for resources

Author: I updated the suggestion to use four backticks on the suggestion block so that the three backticks within the suggestion wouldn't break the suggestion block.

- **KubeSpawner and deletion of PVCs** ([jupyterhub#3337](https://github.com/jupyterhub/jupyterhub/pull/3337), [kubespawner#475](https://github.com/jupyterhub/kubespawner/pull/475))

  Deleting a user in JupyterHub's admin interface (/hub/admin) or removing a
  named server will now lead to the deletion of the user's or named server's
  dynamically created PVC resource if there was one.

  To opt out of this behavior and retain the current behavior where dynamically

>   created PVC resources will remain, configure KubeSpawner's `delete_pvc`
>   configuration to be false.

Reviewer 2: ```suggestion
  created PVC resources will remain, set `KubeSpawner.delete_pvc` to `false`.
```

  ```yaml
  hub:
    config:
      KubeSpawner:
        delete_pvc: false
  ```

  Note that this feature relies on both KubeSpawner 1.0.0+ and JupyterHub 1.4.1+

>   which is now used as part of this release.

Reviewer 2: ```suggestion
  which are included in this release.
```

        cpu: 0
        memory: 0
    hook:
      resources:
        requests:
          cpu: 0
          memory: 0
  ```

- **KubeSpawner and deletion of PVCs** ([jupyterhub#3337](https://github.com/jupyterhub/jupyterhub/pull/3337), [kubespawner#475](https://github.com/jupyterhub/kubespawner/pull/475))

  Deleting a user in JupyterHub's admin interface (/hub/admin) or removing a
  named server will now lead to the deletion of the user's or named server's
  dynamically created PVC resource if there was one.

  To opt out of this behavior and retain the current behavior where dynamically
  created PVC resources will remain, set `KubeSpawner.delete_pvc` to `false`.

  ```yaml
  hub:
    config:
      KubeSpawner:
        delete_pvc: false
  ```

  Note that this feature relies on both KubeSpawner 1.0.0+ and JupyterHub 1.4.1+
  which are included in this release.

- **hub.existingSecret is reworked** ([#2042](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/2042))

  See [the documentation](http://z2jh.jupyter.org/en/latest/resources/reference.html#hub-existingsecret) and [pull request #2042](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/2042) for more details.

> 

Reviewer 2: ```suggestion

 - **configurable-http-proxy statsd metrics replaced by prometheus metrics** ([#2231](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/2231))
 
   [statsd metrics have been removed and support for Prometheus metrics added](https://github.com/jupyterhub/configurable-http-proxy/pull/314) in configurable-http-proxy.
```

Author: The software supports it, if enabled. But, the Helm chart doesn't expose it properly yet etc so there are some struggles still.

I opened #2232 to chart integration with the the CHP metrics to properly expose it to prometheus running in a k8s cluster.

> 

Reviewer 2: ```suggestion

 - **configurable-http-proxy statsd metrics removed** ([#2231](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/2231))
 
   [statsd metrics have been removed](https://github.com/jupyterhub/configurable-http-proxy/pull/314) in configurable-http-proxy. This will only affect administrators who have overridden the CHP command line arguments as statsd is not supported in the Helm chart. Support for Prometheus metrics will be added in a future release.
```
How about this, so we don't block the release?

| [ldapauthenticator](https://github.com/jupyterhub/ldapauthenticator)             | 1.3.2             | 1.3.2             | [Changelog](https://github.com/jupyterhub/ldapauthenticator/blob/master/CHANGELOG.md)       | Run in the `hub` pod               |
| [ltiauthenticator](https://github.com/jupyterhub/ltiauthenticator)               | 1.0.0             | 1.0.0             | [Changelog](https://github.com/jupyterhub/ltiauthenticator/blob/master/CHANGELOG.md)        | Run in the `hub` pod               |
| [nativeauthenticator](https://github.com/jupyterhub/nativeauthenticator)         | 0.0.6             | 0.0.7             | [Changelog](https://github.com/jupyterhub/nativeauthenticator/blob/master/CHANGELOG.md)     | Run in the `hub` pod               |
| [jupyterhub-idle-culler](https://github.com/jupyterhub/jupyterhub-idle-culler)   | 1.0               | 1.1               | -                                                                                           | Run in the `hub` pod               |

> | [configurable-http-proxy](https://github.com/jupyterhub/configurable-http-proxy) | 4.2.2             | 4.3.2             | [Changelog](https://github.com/jupyterhub/configurable-http-proxy/blob/master/CHANGELOG.md) | Run in the `proxy` pod             |

Reviewer 2: ```suggestion
| [configurable-http-proxy](https://github.com/jupyterhub/configurable-http-proxy) | 4.2.2             | 4.4.0             | [Changelog](https://github.com/jupyterhub/configurable-http-proxy/blob/master/CHANGELOG.md) | Run in the `proxy` pod             |
```
